pkg/webfinger/client/client.go

```diff
@@ -141,6 +107,76 @@ func (c *Client) HasSupportedLedgerType(domain string) (bool, error) {
 	return contains(supportedLedgerTypes, lt), nil
 }
 
+// ResolveWebFingerResource attempts to resolve the given WebFinger resource from domainWithScheme.
+func (c *Client) ResolveWebFingerResource(domainWithScheme, resource string) (restapi.JRD, error) {
+	webFingerURL := fmt.Sprintf("%s/.well-known/webfinger?resource=%s", domainWithScheme, resource)
+
+	req, err := http.NewRequest(http.MethodGet, webFingerURL, nil)
+	if err != nil {
+		return restapi.JRD{},
+			fmt.Errorf("failed to create new request for WebFinger URL [%s]: %w", webFingerURL, err)
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return restapi.JRD{}, fmt.Errorf("failed to get response (URL: %s): %w", webFingerURL, err)
+	}
+
+	defer func() {
+		err = resp.Body.Close()
+		if err != nil {
+			logger.Errorf("failed to close response body after getting WebFinger response: %s", err.Error())
+		}
+	}()
+
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return restapi.JRD{}, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	if resp.StatusCode == http.StatusNotFound {
+		return restapi.JRD{}, model.ErrResourceNotFound
+	} else if resp.StatusCode != http.StatusOK {
+		return restapi.JRD{}, fmt.Errorf("received unexpected status code. URL [%s], "+
+			"status code [%d], response body [%s]", webFingerURL, resp.StatusCode, string(respBytes))
+	}
+
+	webFingerResponse := restapi.JRD{}
+
+	err = json.Unmarshal(respBytes, &webFingerResponse)
+	if err != nil {
+		return restapi.JRD{}, fmt.Errorf("failed to unmarshal WebFinger response: %w", err)
+	}
+
+	return webFingerResponse, nil
+}
+
+// GetWebCASURL gets the WebCAS URL for cid from domainWithScheme using WebFinger.
+func (c *Client) GetWebCASURL(domainWithScheme, cid string) (*url.URL, error) {
+	webFingerResponse, err := c.ResolveWebFingerResource(domainWithScheme,
```

Comment:
We should add cache for webcas URL (separate PR)

Reply:
Created an issue and added a TODO